cmd/jujud/agent/machine.go

```diff
@@ -1194,16 +1194,20 @@ func openStatePool(
 	}
 	defer session.Close()
 
-	var credService stateenvirons.CredentialService
+	var (
+		credService  stateenvirons.CredentialService
+		cloudService stateenvirons.CloudService
+	)
 	if serviceFactory != nil {
 		credService = serviceFactory.Credential()
+		cloudService = serviceFactory.Cloud()
```

Comment:
This pattern will go away once we have a real service factory in all the test suites.